omod/src/main/java/org/openmrs/module/appointmentscheduling/rest/resource/openmrs1_9/AppointmentResource1_9.java

```diff
@@ -127,7 +128,11 @@ public Appointment save(Appointment appointment) {
 	protected Appointment save(Appointment appointment, Boolean allowOverbook) {
 		if (appointment.getId() != null) {
 			// existing appointments get updated
-			return Context.getService(AppointmentService.class).saveAppointment(appointment);
+			AppointmentStatusHistory statusHistory = Context.getService(AppointmentService.class).getMostRecentAppointmentStatusHistory(appointment);
+			if (appointment.getStatus() != statusHistory.getStatus()) {
+				Context.getService(AppointmentService.class).changeAppointmentStatus(appointment, appointment.getStatus());
+			}

```

Comment:
Shouldn't there be an else to call "saveAppointment" here if there is no status change?

Reply:
got it